### PR TITLE
Fixes for base64 commands

### LINKED ISSUE #43: base64encode command will change multibyte string into unicode string first, and then do the encoding
for example(Chinese words):

```plain
百度
```
command: `StringEncoding: Base64 Decode`, got:

```
XHU3NjdlXHU1ZWE2
```
actually it's the result of `\u767e\u5ea6`, the right result is below:

```
55m+5bqm
```

### PR DESCRIPTION
**Pull request by deathaxe**
Fixes #43

1. ensure correct padding when decoding base64 strings
2. ensure correct encodings
   - base64 encoded values are always ascii
   - use utf-8 to encode and decode content

### CODE REVIEW

File: string_encode.py [hunk 1]
```diff
@@ -101,12 +101,19 @@ def run(self, edit, **kwargs):
...
 
 class Gzip64DecodeCommand(StringEncode):
 
     def encode(self, text):
-        return gzip.decompress(base64.b64decode(text.rstrip('=') + '===')).decode('utf-8')
+        value = bytearray(text, 'ascii')
+        mod = len(value) % 4
+        if mod == 3:
+            value.extend(b"=")
+        elif mod == 2:
```
Comment (colinta) [inline string_encode.py:113]: No case needed for `mod == 1`?

Comment (deathaxe) [reply string_encode.py:113]: 4 byte of base64 string represent 3 byte unencoded data. A single `=` indicates 2 bytes and `==` indicates 1 byte of unencoded data. Hence `mod == 1` most likely indicates invalid base64 string or already over-padded string.

Comment (colinta) [reply string_encode.py:143]: Ahh I see I was interpreting this the other way around, I was reviewing by phone and assumed this was padding the encode step (which also seemed unnecessary but I was putting that on pause).

I'm at the computer, and yes this works as intended when decoding data that *isn't* correctly padded. The only time the `mod == 1` comes up is when I try to decode *in-between* the byte.

```
dGVzdA==  => test
dGVzdA=   => test
dGVzdA    => test
dGVzd     => invalid
````

**Comment by colinta** [comment]
Merged, added a `pad64` helper.

**Pull request merged**